airbyte-integrations/bases/base-java/src/test/java/io/airbyte/integrations/destination_async/PartialAirbyteMessageTest.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination_async;
+
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.integrations.destination_async.partial_messages.PartialAirbyteMessage;
+import io.airbyte.protocol.models.AirbyteStateMessage;
+import io.airbyte.protocol.models.AirbyteStreamState;
+import io.airbyte.protocol.models.StreamDescriptor;
+import io.airbyte.protocol.models.v0.AirbyteMessage;
+import io.airbyte.protocol.models.v0.AirbyteRecordMessage;
+import java.time.Instant;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class PartialAirbyteMessageTest {
+
+  @Test
+  void testDeserializeRecord() {
+    final long emittedAt = Instant.now().toEpochMilli();
+    final var serializedRec = Jsons.serialize(new AirbyteMessage()
+        .withType(AirbyteMessage.Type.RECORD)
+        .withRecord(new AirbyteRecordMessage()
+            .withStream("users")
+            .withNamespace("public")
+            .withEmittedAt(emittedAt)
+            .withData(Jsons.jsonNode("data"))));
+
+    final var rec = Jsons.tryDeserialize(serializedRec, PartialAirbyteMessage.class).get();
```

Comment:
@airbytehq/destinations this is the specific metadata that will be parsed for an `AirbyteRecord` in the async world